nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core-api/src/main/java/org/apache/nifi/groups/FlowSynchronizationCallback.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.groups;
+
+import org.apache.nifi.connectable.Port;
+import org.apache.nifi.controller.ProcessorNode;
+import org.apache.nifi.controller.ReportingTaskNode;
+import org.apache.nifi.controller.service.ControllerServiceNode;
+
+public interface FlowSynchronizationCallback {
```

Comment:
I'm OK with the concept of listening for these events. But I'm not sure that `FlowSynchronizationCallback` is the right abstraction here - er, rather, the right name.
When you call it a callback, it gives the impression that it's some action that needs to be taken after a flow is synchronized. Here, there's no action that needs to be taken in order to complete the synchronization. Instead, it's more of an event listener. And the methods have nothing to do with Flow Synchronization but rather a change of components' scheduled states.
Perhaps it makes more sense to refer to this as a ScheduleStateChangeListener or something like that?

Reply:
Good call -- I'll go with `ScheduleStateChangeListener`